can/io/asc.py

```diff
@@ -38,15 +44,32 @@ def __init__(self, file):
                      read mode, not binary read mode.
         """
         super().__init__(file, mode="r")
-
-    @staticmethod
-    def _extract_can_id(str_can_id):
+        self.base = 16
+        self.absolute_timestamps = False
+        self._parse_header()
+
+    def _parse_header(self):
+        """Parse the header for information about base and timestamps."""
+        base_regex = r"base\s+(?P<base>\w+)\s+timestamps\s+(?P<timestamps>\w+)"
+        for i in range(10):
+            line = self.file.readline()
+            m = re.search(base_regex, line)
+            if m:
+                base = m.group('base')
+                if base not in BASES:
+                    raise ASCParseError("Support for base %s not implemented" % base)
+                self.base = BASES[base]
+                self.absolute_timestamps = m.group('timestamps') == 'absolute'
+                break
+        self.file.seek(0)
```

Comment:
Is it really necessary to seek? Couldn’t we just parse the header until we hit Begin Triggerblock and then continue with the actual log entries? The reason I’m asking is because there might be cases where one would like to stream the log over e.g. HTTP or something and then it would not be possible to seek.